libraries/AP_RangeFinder/AP_RangeFinder_Lua.cpp

```diff
@@ -0,0 +1,66 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "AP_RangeFinder_Lua.h"
+#include <AP_HAL/AP_HAL.h>
+
+#if AP_SCRIPTING_ENABLED
+
+const AP_Param::GroupInfo AP_RangeFinder_Lua::var_info[] = {
+    // @Param: PRIMARY
+    // @DisplayName: Set Lua Script RangeFinder as Primary Source
+    // @Description: Set 1 to use Lua Script Driver as a Primary RangeFinder (even if the data is out of range)
+    // @Values: 0:Normal Sensor , 1: Primary Sensor
+    // @User: Advanced
+    AP_GROUPINFO("PRIMARY", 25, AP_RangeFinder_Lua, _primary, 0),
```

Comment:
this should be a parameter created in the Lua script
and possibly a set_primary() binding in lua